app/src/main/java/com/futurumgame/base/serialization/parsing/ResourceParseRule.java

```diff
@@ -0,0 +1,53 @@
+package com.futurumgame.base.serialization.parsing;
+
+import com.futurumgame.base.additionalDatatypes.Units;
+import com.futurumgame.base.util.CollectionHelper;
+import com.futurumgame.base.enums.Separator;
+import com.futurumgame.base.exceptions.UnitsFormatException;
+import com.futurumgame.base.resources.Resource;
+import com.futurumgame.base.util.Logger;
+import com.futurumgame.base.util.ResourceMapping;
+
+public class ResourceParseRule extends BaseParseRule<Resource> {
+
+    public ResourceParseRule() {
+    }
+
+    @Override
+    public ParseResult<Resource> next(String string) {
+        super.next(string);
+        if (getReadChars().isEmpty()) {
+            return ParseResult.failResult();
+        }
+        String[] data = getReadChars().split(Separator.DataStructureSeparator.getSeparator());
```

Comment:
Maybe consider saving the value of getReadChars if you use it that often....

Reply:
saved as an encapsuled member variable, thats why there is the use of the getter method

Reply:
kk